src/__tests__/poller.test.ts

```diff
@@ -0,0 +1,135 @@
+import Poller from '../poller'
+import type { Options } from '../types'
+import { getRandom } from '../utils'
+
+jest.useFakeTimers()
+
+jest.mock('../utils.ts', () => ({
+  getRandom: jest.fn()
+}))
+
+interface PollerArgs {
+  fetchFlags: jest.MockedFunction<() => Promise<any>>
+  configurations: Partial<Options>
+  pollInterval: number
+  maxAttempts: number
+}
+
+interface TestArgs {
+  delayFunction: jest.MockedFunction<() => number>
+  delayMs: number
+}
+
+const getPoller = (overrides: Partial<PollerArgs> = {}): { poller: Poller; pollerArgs: PollerArgs } => {
```

Comment:
Instead of passing back what this used, the caller should be responsible for passing in what they need to observe. That way you can eliminate the `pollerArgs` entirely. If a test needs to observe `fetchFlags`, then it can create it itself and pass it in as an override. If it doesn't need to observe it, then it leaves it to the defaults.

Reply:
AHHH - that was the solution. Awesome - I'll update the code.